### PR TITLE
Improve `Performance/MapCompact` to handle more safe navigation calls

### PR DESCRIPTION
**Pull request by fatkodima**
Fixes https://github.com/rubocop/rubocop-performance/issues/389.

### CODE REVIEW

File: lib/rubocop/cop/performance/map_compact.rb [hunk 1]
```diff
@@ -39,11 +39,11 @@ class MapCompact < Base
 
         def_node_matcher :map_compact, <<~PATTERN
           {
-            (send
+            ({send csend}
```
Comment (koic) [inline lib/rubocop/cop/performance/map_compact.rb:42]:
```suggestion
            (call
```

File: lib/rubocop/cop/performance/map_compact.rb [hunk 2]
```diff
@@ -39,11 +39,11 @@ class MapCompact < Base
...
               $(call _ {:map :collect}
                 (block_pass
                   (sym _))) _)
-            (send
+            ({send csend}
```
Comment (koic) [inline lib/rubocop/cop/performance/map_compact.rb:46]:
```suggestion
            (call
```

**Comment by koic** [comment]
CI has been failed. Can you rebase with the latest master branch?

**Comment by fatkodima** [comment]
Updated.

**Comment by koic** [comment]
Thanks!

**Pull request merged**